storage/storage.go

```diff
@@ -120,13 +120,26 @@ func NewClient(ctx context.Context, opts ...option.ClientOption) (*Client, error
 		opts = append(opts, internaloption.WithDefaultEndpoint("https://storage.googleapis.com/storage/v1/"))
 		opts = append(opts, internaloption.WithDefaultMTLSEndpoint("https://storage.mtls.googleapis.com/storage/v1/"))
 	} else {
-		scheme = "http"
-		readHost = host
+		// Add scheme for user if not supplied in STORAGE_EMULATOR_HOST
+		var hostURL *url.URL
+		if strings.Contains(host, "://") {
```

Comment:
Is this check necessary? Can you just parse the supplied URL and then add the scheme if it's not present already?

Reply:
Added a comment to clarify this - if the scheme is not supplied but the host is, then `url.Parse()` has undefined behaviour